src/Nethermind/Nethermind.Serialization.Rlp/AccountDecoder.cs

```diff
@@ -53,8 +52,12 @@ public Keccak DecodeStorageRootOnly(RlpStream rlpStream)
             UInt256 balance = rlpStream.DecodeUInt256();
             Keccak storageRoot = DecodeStorageRoot(rlpStream);
             Keccak codeHash = DecodeCodeHash(rlpStream);
-            Account account = new(nonce, balance, storageRoot, codeHash);
-            return account;
+            if (ReferenceEquals(storageRoot, Keccak.EmptyTreeHash) && ReferenceEquals(codeHash, Keccak.OfAnEmptyString))
```

Comment:
So... why not put this logic in constructor?

Reply:
We know we can get away with a simple reference equals check here rather than an equality check; ideally could link it to the CompressingDb and not even add these to the rlp and just create the account rather than expanding them in array 🤔